### PR TITLE
Trait Display for Instructions

### LINKED ISSUE #8: Missing Trait Display for Instructions
The decoded instruction prints in debug mode.

### PR DESCRIPTION
**Pull request by jcardoso13**
Correct Instruction print
Closes #8

### CODE REVIEW

File: src/riscv/decoder.rs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 use super::isa::{InstrType, RVT};
+use riscv::labels::*;
 use std::cmp::PartialEq;
-
-#[derive(Debug, Eq)]
+use std::fmt::{self, Display, Formatter};
```
Comment (raffitz) [inline src/riscv/decoder.rs]: Can I be pedantic and ask for a line break between the uses and the Instruction declaration?

File: src/riscv/decoder.rs [hunk 2]
```diff
@@ -1,7 +1,8 @@
 use super::isa::{InstrType, RVT};
+use riscv::labels::*;
 use std::cmp::PartialEq;
-
-#[derive(Debug, Eq)]
```
Comment (raffitz) [inline src/riscv/decoder.rs]: Debug should be maintained

File: src/riscv/isa.rs [hunk 3]
```diff
@@ -83,6 +89,57 @@ impl RVT {
...
 
+impl Display for InstrType {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "{} ", self.instr_op)
+    }
+}
+
+impl Display for RV32I {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match *self {
```
Comment (raffitz) [inline src/riscv/isa.rs]: I believe you could achieve the same with:
```rust
let stuff = match *self {
    RV32I::ADD => "add",
    RV32I::ADDI => "addi",
...
}
write!(f,stuff);
```

**Comment by lfiolhais** [comment]
Superceded by #12

**Pull request closed without merging**